src/chunk.c

```diff
@@ -921,6 +1013,77 @@ append_chunk_oid(ChunkScanCtx *scanctx, Chunk *chunk)
 	return true;
 }
 
+/* A utility function to check that the argument type passed to be
+ * used/compared with a hypertable time column is valid
+ * The argument is valid if
+ * 	-	it is an INTEGER type and time column of hypertable is also INTEGER
+ * 	-	it is an INTERVAL and time column of hypertable is time or date
+ * 	-	it is the same as time column of hypertable
+ */
+static void
+time_dim_typecheck(Oid arg_type, Oid time_column_type, char *caller_name)
+{
+	AssertArg(arg_type != InvalidOid);
+	AssertArg(IS_VALID_OPEN_DIM_TYPE(time_column_type));
+
+	if (IS_INTEGER_TYPE(time_column_type) && IS_INTEGER_TYPE(arg_type))
+		return;
+
+	if (arg_type == INTERVALOID)
```

Comment:
Should we give a deprecation warning when `drop_chunks` or `show_chunks` is used with `INTERVAL` argument type? 
During discussion of `show_chunks` api it has come up several times. 
Main arguments for deprecating
* It is as simple as passing `now()- interval` instead of `interval` when the option is not supported but makes it more explicit
* It becomes less clear what exactly it means when there also is a `newer_than` interval (is it now() - interval or now() + interval ?)

@mfreed @cevian @erimatnor @JLockerman @amytai